{"name": "TC004 - Login with Invalid Credentials", "status": "failed", "statusDetails": {"message": "Failed: Login failed - URL did not change to dashboard", "trace": "setup = <selenium.webdriver.edge.webdriver.WebDriver (session=\"ba652fa4223a2660e7fc6ba6d0c68a96\")>\n\n    @allure.feature(\"XEED_Ventures_Login_Page\")\n    @allure.title(\"TC004 - Login with Invalid Credentials\")\n    @allure.description(\"\"\"\n    **Test Scenario:**\n    Attempt to log in using an invalid credentials\n    Username: Ohitr\n    Password: 9902@KM\n    \n    **Expected Result:**\n    User should not be able to log in.\n    \n    **How it is evaluated:**\n    This is verified by checking the URL change; if the login is successful, the URL changes to `/dashboard`.\n    \"\"\")\n    \n    def test_site(setup):\n        driver = setup\n        wait = WebDriverWait(driver, 10)\n    \n        with allure.step(\"Login Page is Opened\"):\n            driver.get(\"http://157.15.202.244:88/authentication/login\")\n    \n        with allure.step(\"Waiting until the username field is displayed\"):\n            element = wait.until(EC.element_to_be_clickable((By.XPATH, \"//input[@id='username']\")))\n            element.click()\n    \n        with allure.step(\"Typing the invalid username\"):\n            element.send_keys(\"Ohitr\")\n    \n        with allure.step(\"Waiting until the password field is displayed\"):\n            element = wait.until(EC.element_to_be_clickable((By.XPATH, \"//input[@id='password']\")))\n            element.click()\n    \n        with allure.step(\"Typing the invalid password\"):\n            element.send_keys(\"9902@KM\")\n    \n        with allure.step(\"Clicking the login button\"):\n            element = wait.until(EC.element_to_be_clickable((By.XPATH, \"//button[normalize-space()='Login']\")))\n            element.click()\n    \n        with allure.step(\"Waiting for the URL to change to dashboard\"):\n            try:\n>               wait.until(EC.url_contains(\"/dashboard\"))\n\nXeed_Login_Page\\Test_TC004.py:49: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"ba652fa4223a2660e7fc6ba6d0c68a96\")>, method = <function url_contains.<locals>._predicate at 0x000001CEA5E0A8E0>\nmessage = ''\n\n    def until(self, method: Callable[[D], Union[Literal[False], T]], message: str = \"\") -> T:\n        \"\"\"Wait until the method returns a value that is not False.\n    \n        Calls the method provided with the driver as an argument until the\n        return value does not evaluate to ``False``.\n    \n        Parameters:\n        -----------\n        method: callable(WebDriver)\n            - A callable object that takes a WebDriver instance as an argument.\n    \n        message: str\n            - Optional message for :exc:`TimeoutException`\n    \n        Return:\n        -------\n        object: T\n            - The result of the last call to `method`\n    \n        Raises:\n        -------\n        TimeoutException\n            - If 'method' does not return a truthy value within the WebDriverWait\n            object's timeout\n    \n        Example:\n        --------\n        >>> from selenium.webdriver.common.by import By\n        >>> from selenium.webdriver.support.ui import WebDriverWait\n        >>> from selenium.webdriver.support import expected_conditions as EC\n    \n        # Wait until an element is visible on the page\n        >>> wait = WebDriverWait(driver, 10)\n        >>> element = wait.until(EC.visibility_of_element_located((By.ID, \"exampleId\")))\n        >>> print(element.text)\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            if time.monotonic() > end_time:\n                break\n            time.sleep(self._poll)\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message:\n\nvenv\\Lib\\site-packages\\selenium\\webdriver\\support\\wait.py:146: TimeoutException\n\nDuring handling of the above exception, another exception occurred:\n\nsetup = <selenium.webdriver.edge.webdriver.WebDriver (session=\"ba652fa4223a2660e7fc6ba6d0c68a96\")>\n\n    @allure.feature(\"XEED_Ventures_Login_Page\")\n    @allure.title(\"TC004 - Login with Invalid Credentials\")\n    @allure.description(\"\"\"\n    **Test Scenario:**\n    Attempt to log in using an invalid credentials\n    Username: Ohitr\n    Password: 9902@KM\n    \n    **Expected Result:**\n    User should not be able to log in.\n    \n    **How it is evaluated:**\n    This is verified by checking the URL change; if the login is successful, the URL changes to `/dashboard`.\n    \"\"\")\n    \n    def test_site(setup):\n        driver = setup\n        wait = WebDriverWait(driver, 10)\n    \n        with allure.step(\"Login Page is Opened\"):\n            driver.get(\"http://157.15.202.244:88/authentication/login\")\n    \n        with allure.step(\"Waiting until the username field is displayed\"):\n            element = wait.until(EC.element_to_be_clickable((By.XPATH, \"//input[@id='username']\")))\n            element.click()\n    \n        with allure.step(\"Typing the invalid username\"):\n            element.send_keys(\"Ohitr\")\n    \n        with allure.step(\"Waiting until the password field is displayed\"):\n            element = wait.until(EC.element_to_be_clickable((By.XPATH, \"//input[@id='password']\")))\n            element.click()\n    \n        with allure.step(\"Typing the invalid password\"):\n            element.send_keys(\"9902@KM\")\n    \n        with allure.step(\"Clicking the login button\"):\n            element = wait.until(EC.element_to_be_clickable((By.XPATH, \"//button[normalize-space()='Login']\")))\n            element.click()\n    \n        with allure.step(\"Waiting for the URL to change to dashboard\"):\n            try:\n                wait.until(EC.url_contains(\"/dashboard\"))\n            except:\n                allure.attach(driver.current_url, name=\"Current URL on failure\", attachment_type=allure.attachment_type.TEXT)\n>               pytest.fail(\"Login failed - URL did not change to dashboard\")\nE               Failed: Login failed - URL did not change to dashboard\n\nXeed_Login_Page\\Test_TC004.py:52: Failed"}, "description": "\n**Test Scenario:**  \nAttempt to log in using an invalid credentials  \nUsername: Ohitr\nPassword: 9902@KM\n\n**Expected Result:**  \nUser should not be able to log in.\n\n**How it is evaluated:**  \nThis is verified by checking the URL change; if the login is successful, the URL changes to `/dashboard`.\n", "steps": [{"name": "Login Page is Opened", "status": "passed", "start": 1749207362098, "stop": 1749207362242}, {"name": "Waiting until the username field is displayed", "status": "passed", "start": 1749207362242, "stop": 1749207364181}, {"name": "Typing the invalid username", "status": "passed", "start": 1749207364181, "stop": 1749207364228}, {"name": "Waiting until the password field is displayed", "status": "passed", "start": 1749207364228, "stop": 1749207364324}, {"name": "Typing the invalid password", "status": "passed", "start": 1749207364324, "stop": 1749207364366}, {"name": "Clicking the login button", "status": "passed", "start": 1749207364366, "stop": 1749207364447}, {"name": "Waiting for the URL to change to dashboard", "status": "failed", "statusDetails": {"message": "Failed: Login failed - URL did not change to dashboard\n", "trace": "  File \"C:\\Users\\sagar\\PycharmProjects\\TestingProject\\Xeed_Login_Page\\Test_TC004.py\", line 52, in test_site\n    pytest.fail(\"Login failed - URL did not change to dashboard\")\n    ~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\sagar\\PycharmProjects\\TestingProject\\venv\\Lib\\site-packages\\_pytest\\outcomes.py\", line 178, in fail\n    raise Failed(msg=reason, pytrace=pytrace)\n"}, "attachments": [{"name": "Current URL on failure", "source": "d0ca598e-aa84-4520-9d9a-19315768edb4-attachment.txt", "type": "text/plain"}], "start": 1749207364447, "stop": 1749207374551}], "parameters": [{"name": "setup", "value": "'edge'"}], "start": 1749207362097, "stop": 1749207374552, "uuid": "8c326193-40d4-4920-af34-773c12d9cae9", "historyId": "8785abd94e2f36beb2df8c453130064d", "testCaseId": "df66409949c786e429fa62b97ff2cb96", "fullName": "Xeed_Login_Page.Test_TC004#test_site", "labels": [{"name": "feature", "value": "XEED_Ventures_Login_Page"}, {"name": "parentSuite", "value": "Xeed_Login_Page"}, {"name": "suite", "value": "Test_TC004"}, {"name": "host", "value": "DESKTOP-0Q1NQ4H"}, {"name": "thread", "value": "40464-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "Xeed_Login_Page.Test_TC004"}]}